### DISCUSSION TITLE
combine two or more DynamicItemDatasets

### DISCUSSION THREAD

**Discussion by cassiotbatista** [speechbrain/speechbrain#1831] [Feature Request]
Hi, 

Is there a way to do that, like [torch's ConcatDataset](https://pytorch.org/docs/stable/data.html#torch.utils.data.ConcatDataset)? Could be useful instead of merging JSON or CSV files before, as pre-processing state. 

Right now I'm solving with subclasses. It works, but it just doesn't feel right. Example:

Suppose I wanted to merge `test-clean` and `test-other` subsets of Librispeech. After [loading CSVs individually](https://github.com/speechbrain/speechbrain/blob/develop/recipes/LibriSpeech/ASR/transformer/train.py#L312), I'd have two `DynamicItemDataset` objects. I could concat them before filtering/sorting, but the return type would be a `ConcatDataset` object which won't allow me to do the filtering/sorting after. So I created a child class which allows me to use the `+` operator on two `DynamicItemDatasets` objects.

```python
class CustomDynamicItemDataset(sb.dataio.dataset.DynamicItemDataset):           
    def __add__(self, x):                                                       
        new_data = {}                                                           
        for d in (self.data, x.data):                                           
            for k, v in d.items():                                              
                if k in new_data.keys():                                        
                    raise ValueError("dup uttids, can't merge!")                
                elif k == "dummy":                                              
                    continue                            
                new_data[k] = v                                             
        return CustomDynamicItemDataset(new_data, [], [])
```

The problem is, for a unknown number of objects to be concatenated, I need to use this work around via a dummy object for `sum()` to work.

```python
dummy = CustomDynamicItemDataset({"dummy": {"dummy": 0}}, [], []) 
test_data = sum([i for k, i in test_datasets.items()], dummy)
test_data = test_data.filtered_sorted(sort_key="duration", reverse=True)  # now it works
```

**Comment by Adel-Moumen** [comment]
If I am correct, @Chaanks you already faced this kind of problem right? How did you solved it please? :-)

**Comment by BrownsugarZeer** [comment]
@Chaanks Are there any details? I also face the same difficulty to combine two dataset.
